pkg/store/event/event.go

```diff
@@ -44,15 +44,114 @@ type eventStore struct {
 	db *dbstore.DB
 }
 
-// TODO: implement core.EventStore interface
-func (e *eventStore) List(context.Context) ([]*core.Event, error) { return nil, nil }
-func (e *eventStore) ListByExperiment(context.Context, string, string) ([]*core.Event, error) {
-	return nil, nil
+// FindPodsByEventID return the list of PodRecords according to the eventID
+func (e *eventStore) FindPodsByEventID(_ context.Context, id uint) ([]*core.PodRecord, error) {
+	pods := make([]*core.PodRecord,0)
+	if err := e.db.Where(
+		"event_id = ?", id).
+		Find(&pods).Error; err != nil && !gorm.IsRecordNotFoundError(err) {
+		return nil, err
+	}
+	return pods, nil
+}
+
+// List returns the list of events
+func (e *eventStore) List(_ context.Context) ([]*core.Event, error) {
+	var resList []core.Event
+	eventList := make([]*core.Event, 0)
+
+	if err := e.db.Find(&resList).Error; err != nil && !gorm.IsRecordNotFoundError(err) {
+		return nil, err
+	}
+
+	for _, et := range(resList) {
+		pods, err:= e.FindPodsByEventID(context.Background(), et.ID)
+		if err != nil {
+			return nil, err
+		}
+		for _, pod := range(pods) {
+			et.Pods = append(et.Pods, pod)
+		}
+		eventList = append(eventList, &et)
+	}
+
+	return eventList, nil
 }
-func (e *eventStore) ListByPod(context.Context, string, string) ([]*core.Event, error) {
-	return nil, nil
+
+// ListByExperiment returns a event list by the name and namespace of the experiment.
+func (e *eventStore) ListByExperiment(_ context.Context, namespace string, experiment string) ([]*core.Event, error) {
+	var resList []core.Event
+	eventList := make([]*core.Event, 0)
+
+	if err := e.db.Where(
+		"namespace = ? and experiment = ? ",
+		namespace, experiment).
+		Find(resList).Error; err != nil && !gorm.IsRecordNotFoundError(err) {
+		return nil, err
+	}
+
+	for _, et := range resList {
+		pods, err:= e.FindPodsByEventID(context.Background(), et.ID)
+		if err != nil {
+			return nil, err
+		}
+		for _, pod := range(pods) {
+			et.Pods = append(et.Pods, pod)
+		}
+		eventList = append(eventList, &et)
+	}
+
+	return eventList, nil
+}
+
+// ListByPod returns the list of events according to the pod
+func (e *eventStore) ListByPod(_ context.Context, namespace string, name string) ([]*core.Event, error) {
+	podRecords := make([]*core.PodRecord, 0)
+
+	if err := e.db.Where(
+		&core.PodRecord{PodName: name, Namespace: namespace}).
+		Find(&podRecords).Error; err != nil && !gorm.IsRecordNotFoundError(err) {
+		return nil, err
+	}
+
+	et := new(core.Event)
+	eventList := make([]*core.Event, 0)
```

Comment:
A small nit.

`eventList := make([]*core.Event, 0, len(podRecords))` would be better